app/controllers/admin_controller.rb

```diff
@@ -181,23 +180,24 @@ def delete_results_data
   end
 
   def create_results
-    @competition = competition_from_params
+    competition = competition_from_params
 
     # Do json analysis + insert record in db, then redirect to check inbox
     # (and delete existing record if any)
-    upload_json_params = params.require(:upload_json).permit(:results_file)
-    upload_json_params[:competition_id] = @competition.id
-    @upload_json = UploadJson.new(upload_json_params)
+    upload_json = UploadJson.new({
+                                   results_file: params.require(:results_file),
+                                   competition_id: competition.id,
+                                 })
+    mark_result_submitted = ActiveRecord::Type::Boolean.new.cast(params.require(:mark_result_submitted))
 
     # This makes sure the json structure is valid!
-    if @upload_json.import_to_inbox
-      @competition.update!(results_submitted_at: Time.now) if @competition.results_submitted_at.nil?
-      flash[:success] = "JSON file has been imported."
-      redirect_to competition_admin_upload_results_edit_path
+    if upload_json.import_to_inbox
+      competition.update!(results_submitted_at: Time.now) if competition.results_submitted_at.nil? && mark_result_submitted
```

Comment:
```suggestion
      competition.touch(:results_submitted_at) if competition.results_submitted_at.nil? && mark_result_submitted
```

Reply:
This is the more idiomatic way to execute "mark current time" on a timestamp column. Note that this does not run model validations except the ones that are intentionally defined as `after_touch` (which I think we don't use in our codebase at all)

So if you have some business logic that complains and doesn't allow `results_submitted_at` to be set based on WRT rules or WCAT rules or other real-world rules (that are defined in our code as Rails validations) then this suggestion is not viable.